includes/Checker/Default_Check_Repository.php

```diff
@@ -99,37 +96,19 @@ public function get_checks( $flags = self::TYPE_ALL, array $check_slugs = array(
 			$checks += $this->runtime_checks;
 		}
 
-		// Filter out the specific check slugs requested.
-		if ( ! empty( $check_slugs ) ) {
-			$checks = array_map(
-				function ( $slug ) use ( $checks ) {
-					if ( ! isset( $checks[ $slug ] ) ) {
-						throw new Exception(
```

Comment:
Before the PR changes( in trunk ), when we ran the check for an unregistered checks, it would display an error in the CLI. However, with the current changes, we no longer receive any errors in this situation. We should determine whether this is the intended behaviour or if there are plans to add error messages for unregistered items later.

<img width="1346" alt="Screenshot 2023-11-01 at 10 40 05 AM" src="https://github.com/WordPress/plugin-check/assets/10103365/38dab21d-bb6a-4451-889e-a9180a9d4649">

Reply:
@mukeshpanchal27 Great point, I made this change consciously though, I just forgot to add a comment about it on the PR. It seemed weird to me to throw an exception in a filtering function, e.g. it could just silently fail.

Now that I think about it again, I am not sure though. Do we want to be strict and fail for such requests? Or do we want to be forgiving? The latter makes it a bit simpler to use, but it could also lead to false expectations, e.g. you want to run only two specific checks one of which e.g. no longer exists or was renamed, but because there's no exception you think the check just passed.

WDYT @mukeshpanchal27 @swissspidy?

Reply:
UX-wise I'd prefer a more strict message and not failing silently, exactly to prevent issues like that with renames etc.

Reply:
I'm also in favor of strict messaging. Without clear messages about renaming or deprecation checks, end users may not fully understand whether their plugin has passed all the necessary checks.

Reply:
@swissspidy @mukeshpanchal27 I have implemented a new `Check_Collection::require()` method for this purpose. This allows us to decouple filtering by specific checks from _requiring_ specific checks to be present.

In our central logic, both functions are now called, i.e. requiring the given checks to be present but then also limiting to only run those. While we could have combined both things into one method, I think the approach of two separate methods allows for more flexibility. For example, there could be a scenario where you really just want to filter on a collection that may _already_ be a subset of checks, so then checking that all slugs are valid may not be applicable.

I've also added proper unit tests for the new class (and as part of that replaced two tests that were previously part of `Default_Check_Repository_Tests`).